applications/affine-finetune-embeddings/model.py

```diff
@@ -0,0 +1,100 @@
+import torch
+import torch.nn.functional as F
+import pytorch_lightning as pl
+import torchmetrics
+
+
+# Similarity Model
+class SimilarityModel(pl.LightningModule):
+    def __init__(self, embedding_size, n_dims, dropout_fraction, lr, use_relu):
+        super(SimilarityModel, self).__init__()
+        self.matrix = torch.nn.Parameter(
+            torch.rand(
+                embedding_size,
+                n_dims,
+            )
+        )
+        torch.nn.init.xavier_uniform_(self.matrix)
+        self.dropout_fraction = dropout_fraction
+        self.lr = lr
+        self.use_relu = use_relu
+
+        self.recall = torchmetrics.Recall(task="binary")
+        self.f1 = torchmetrics.F1Score(num_classes=2, task="binary")
+        self.precision = torchmetrics.Precision(task="binary")
+        self.acc = torchmetrics.Accuracy(task="binary")
+        self.auc = torchmetrics.AUROC(task="binary")
+        self.save_hyperparameters()
+
+    def forward(self, embedding_1, embedding_2):
+        e1 = F.dropout(embedding_1, p=self.dropout_fraction)
```

Comment:
reminder to attach an embedding model so we can pass in text rather than embeddings.

Reply:
sounds good, gonna refactor the model so that it takes in text instead of embedding by using the attached embedding model as the base